genie-agent/src/main/java/com/netflix/genie/agent/cli/CommandsConfig.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.cli;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+class CommandsConfig {
```

Comment:
Since all this code is in one package I think we can get away from the `@Configuration` classes if we want and just go with annotating classes with `@Component`, `@Service` whatever to save on boilerplate.

Reply:
Ok, although as beginner to spring and this codebase it was useful to look for beans in `*Config` files, a bit more centralized, rather than looking for a class of who-knows-what type across the whole codebase.

Reply:
Fair. The IDE would certainly help you discover them via the Spring Context but I can see your point here.